resonantLimits/write_res_card.py

```diff
@@ -454,13 +474,18 @@ def writeCard(backgrounds, signals, select, varfit, regions=()):
 if not opt.isResonant:
     for sel in selections:
         for ireg in range(len(regions)):
-            card = writeCard(backgrounds, signals, sel, opt.var, regions[ireg])
+            writeCard(backgrounds, signals, sel, opt.var, regions[ireg])
 else:
     for sel in selections:
-        for ireg in range(len(regions)):
-            for sig in signals:
-                sigmass = int(sig.replace('ggFRadion', ''))
-                if 'boosted' in sel and sigmass<301:
-                    print('Not generating card for {} in boosted category'.format(sig))
-                else:
-                    card = writeCard(backgrounds, [sig], sel, opt.var, regions[ireg])
+        if 'ttCR_invMcut' in selections:
+            # the signal is passed for convenience; it should be zero
```

Comment:
I'm not sure I understand this; can't you pass an empty array or something like this instead?

Reply:
It turns out I was using a random signal sample to fill the corresponding location in the datacard. Since I am using the options ```--setParameters r=0.00 --freezeParameters r --rMin 0 --rMax 0 --redefineSignalPOI rate_TT```, the TTbar MC is nevertheless considered the signal in the CR fits, and the rate form the (randomly selected) signal sample is set to zero (the ```r``` parameter in combine, often denoted by ```μ```). In conclusion, the fit seems to be correct.
However, to avoid confusion, I've added some options that allow to pass an empty list to the ```signals``` argument of ```writeCard()```. When this is done a dummy signal is added with rate strictly equal to zero. It is then clear from reading the datacard that the fit does not take the signal into account. So far we had to inspect the command used to create the workspace to be sure the signal was correctly ignored.
I'm attaching the output datacard in case someone wants to cross-check (the commands to run the fit can be found under ```scripts/ttSF_fit.sh```): [hhres_UL18_inclusive_ttCR_invMcut_13TeV.NOSIGNAL.txt](https://github.com/LLRCMS/KLUBAnalysis/files/12259851/hhres_UL18_inclusive_ttCR_invMcut_13TeV.NOSIGNAL.txt)